File: qc-core/src/main/java/com/eager/questioncloud/domain/userquestion/UserQuestionRepository.java
package com.eager.questioncloud.domain.userquestion;

import com.eager.questioncloud.domain.question.QuestionFilter;
import java.util.List;

public interface UserQuestionRepository {
    List<UserQuestion> saveAll(List<UserQuestion> userQuestionLibraries);

    Boolean isOwned(Long userId, List<Long> questionIds);

    Boolean isOwned(Long userId, Long questionId);

    List<UserQuestionDetail> getUserQuestions(QuestionFilter questionFilter);

    int countUserQuestions(QuestionFilter questionFilter);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/userquestion/infrastructure/repository/UserQuestionRepository.kt
package com.eager.questioncloud.core.domain.userquestion.infrastructure.repository;

import com.eager.questioncloud.core.domain.question.common.QuestionFilter;
import com.eager.questioncloud.core.domain.userquestion.dto.UserQuestionDetail;
import com.eager.questioncloud.core.domain.userquestion.model.UserQuestion;
import java.util.List;

public interface UserQuestionRepository {
    List<UserQuestion> saveAll(List<UserQuestion> userQuestionLibraries);

    Boolean isOwned(Long userId, List<Long> questionIds);

    Boolean isOwned(Long userId, Long questionId);

    List<UserQuestionDetail> getUserQuestions(QuestionFilter questionFilter);

    int countUserQuestions(QuestionFilter questionFilter);

    void deleteAllInBatch();
}
